src/libstd/fs.rs

```diff
@@ -1724,6 +1734,23 @@ mod tests {
         assert!(canary.exists());
     }
 
+    // FIXME(#12795) depends on lstat to work on windows
+    #[cfg(not(windows))]
```

Comment:
Yes, that's fine for now; once this is merged, you can file a followup issue about adding Windows tests.